pkg/k8shandler/forwarding.go

```diff
@@ -59,28 +60,32 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 		}
 	}
 
-	generator, err := forwarding.NewConfigGenerator(
-		clusterRequest.Cluster.Spec.Collection.Logs.Type,
-		clusterRequest.includeLegacyForwardConfig(),
-		clusterRequest.includeLegacySyslogConfig(),
-		clusterRequest.useOldRemoteSyslogPlugin(),
-	)
+	op := generator.Options{}
+	if clusterRequest.includeLegacyForwardConfig() {
+		op[generator.IncludeLegacyForwardConfig] = ""
+	}
+	if clusterRequest.includeLegacySyslogConfig() {
+		op[generator.IncludeLegacySyslogConfig] = ""
+	}
+	if clusterRequest.useOldRemoteSyslogPlugin() {
+		op[generator.UseOldRemoteSyslogPlugin] = ""
+	}
 
+	err = fluentd.Verify(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
 	if err != nil {
-		log.Error(err, "Unable to create collector config generator")
+		log.Error(err, "Unable to generate log configuration")
 		return "",
 			clusterRequest.UpdateCondition(
 				logging.CollectorDeadEnd,
-				"Unable to generate collector configuration",
-				"No defined logstore destination",
+				"ClusterLogForwarder input validation error",
+				"ClusterLogForwarder input validation error",
 				corev1.ConditionTrue,
 			)
 	}
 
-	clfSpec := &clusterRequest.ForwarderSpec
-	fwSpec := clusterRequest.Cluster.Spec.Forwarder
-
-	generatedConfig, err := generator.Generate(clfSpec, clusterRequest.OutputSecrets, fwSpec)
+	g := generator.MakeGenerator()
+	conf_sections := fluentd.Conf(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
```

Comment:
Rename to camelCase which (I think) is general convention for golang

Reply:
updated